src/Flex.php

```diff
@@ -659,7 +659,7 @@ private function populateRepoCacheDir()
                 continue;
             }
 
-            if (!preg_match('#^https\?://#', $repo['url'])) {
```

Comment:
why was it matching strings starting with `https?://` ? Wasn't the escaping of the  `?` in the regex done by mistake ?

Reply:
not a mistake, but a special case that was not handled properly, see:
https://github.com/composer/composer/blob/master/src/Composer/Repository/ComposerRepository.php#L74

Reply:
permalink for above comment:
- https://github.com/composer/composer/blob/master@%7B2018-07-27T10:39:09Z%7D/src/Composer/Repository/ComposerRepository.php#L74